asserts/constraint.go

```diff
@@ -218,15 +219,23 @@ var (
 )
 
 func compileEvalAttrMatcher(cc compileContext, s string) (attrMatcher, error) {
-	if len(cc.opts.allowedOperations) == 0 {
-		return nil, fmt.Errorf("cannot compile %q constraint %q: no $OP() constraints supported", cc, s)
+	if len(cc.opts.allowedOperations) == 0 && len(cc.opts.allowedRefs) == 0 {
+		return nil, fmt.Errorf("cannot compile %q constraint %q: no $OP() or $REF constraints supported", cc, s)
 	}
+
+	if strutil.ListContains(cc.opts.allowedRefs, s[1:]) {
+		return evalAttrMatcher{op: s[1:]}, nil
```

Comment:
It seems a bit strange to me to call the ref an op, perhaps there's a more general name that could be used here to apply to both?

Reply:
This has been changed now, since there is a matcher specifically for the reference